src/libraries/System.Private.CoreLib/src/System/Guid.cs

```diff
@@ -909,75 +909,75 @@ private static ReadOnlySpan<char> EatAllWhitespace(ReadOnlySpan<char> str)
         // Returns an unsigned byte array containing the GUID.
         public byte[] ToByteArray()
         {
-            var g = new byte[16];
+            var bytes = new byte[16];
             if (BitConverter.IsLittleEndian)
             {
-                MemoryMarshal.TryWrite(g, in this);
+                MemoryMarshal.Write(bytes, in this);
             }
             else
             {
                 // slower path for BigEndian
                 Guid guid = new Guid(MemoryMarshal.AsBytes(new ReadOnlySpan<Guid>(in this)), false);
-                MemoryMarshal.TryWrite(g, in guid);
+                MemoryMarshal.Write(bytes, in guid);
             }
-            return g;
+            return bytes;
         }
 
 
         // Returns an unsigned byte array containing the GUID.
         public byte[] ToByteArray(bool bigEndian)
         {
-            var g = new byte[16];
+            var bytes = new byte[16];
             if (BitConverter.IsLittleEndian != bigEndian)
             {
-                MemoryMarshal.TryWrite(g, in this);
+                MemoryMarshal.Write(bytes, in this);
             }
             else
             {
                 // slower path for Reverse
                 Guid guid = new Guid(MemoryMarshal.AsBytes(new ReadOnlySpan<Guid>(in this)), bigEndian);
-                MemoryMarshal.TryWrite(g, in guid);
+                MemoryMarshal.Write(bytes, in guid);
             }
-            return g;
+            return bytes;
         }
 
         // Returns whether bytes are successfully written to given span.
         public bool TryWriteBytes(Span<byte> destination)
         {
-            if (destination.Length < 16)
+            if ((uint)destination.Length < 16)
```

Comment:
Jit definitely shouldn't need this, it should be aware that Span.Length is never negative

Reply:
There is an unsigned length check in `MemoryMarshal.Write`:

https://github.com/dotnet/runtime/blob/0378936909464c84cf207ffd1a21efa474fc34c0/src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/MemoryMarshal.cs#L514

The JIT won't elide the check in the case where one length check is signed, the other unsigned:

https://csharp.godbolt.org/z/WTj89j5Mn

Reply:
The check in `MemoryMarshal.Write` could be made signed, but then there could be a problem calling `Write` with a sliced span, due to the unsigned conditionals in `Span.Slice`.

Reply:
I don't see bound checks in your godbolt link, I do see two similar conditions that jit didn't remove - it's a different issue that we should also fix. Like I said previously, if JIT fails to fix an obvious case - we should file an issue instead of applying hacks in the codebase. These hacks still make sense for cases where JIT likely has no ability to know that certain external argument is never negative.

Reply:
> I don't see bound checks

What I mean is that we currently have two different algorithms for bound checks specifically and a general redundant branch removal

Reply:
@AndyAyersMS you might be interested in the godbolt link above (https://csharp.godbolt.org/z/WTj89j5Mn) for RBO/JumpThreading missing opportunity

Reply:
Btw, we have an issue to remove existing `(uint)` hacks - https://github.com/dotnet/runtime/issues/67044 so we shouldn't do the opposite

Reply:
https://github.com/xtqqczze/dotnet-runtime/commit/152074163ad1f19ec93ad3de4bbd5350d271901e